src/main/host/syscall/handler/timerfd.rs

```diff
@@ -54,17 +55,20 @@ impl SyscallHandler {
             .register_descriptor(desc)
             .or(Err(Errno::ENFILE))?;
 
-        log::trace!("timerfd_create() returning fd {}", fd);
+        log::trace!("timerfd_create() returning fd {fd}");
 
-        Ok(fd.val().into())
+        i32::try_from(fd.val()).or(Err(Errno::ENFILE.into()))
```

Comment:
I think it would be better to either panic here, or unregister the fd. Otherwise this will leak a timerfd that will sit in the descriptor table forever.